src/main/java/uk/gov/hmcts/reform/hmc/service/common/OverrideAuditService.java

```diff
@@ -0,0 +1,161 @@
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+
+import static org.springframework.data.util.CastUtils.cast;
+import static uk.gov.hmcts.reform.hmc.constants.Constants.OVERRIDE_URL;
+import static uk.gov.hmcts.reform.hmc.constants.Constants.OVERRIDE_URL_EVENT;
+import static uk.gov.hmcts.reform.hmc.data.SecurityUtils.SERVICE_AUTHORIZATION;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class OverrideAuditService {
+
+    private static final String BEARER_PREFIX = "Bearer ";
+    private static final String PARAM_ATTRIBUTE =
+        "org.springframework.web.servlet.HandlerMapping.uriTemplateVariables";
+
+    private final HearingStatusAuditRepository hearingStatusAuditRepository;
+    private final LinkedHearingStatusAuditRepository linkedHearingStatusAuditRepository;
+    private final UrlManager roleAssignmentUrlManager;
+    private final UrlManager dataStoreUrlManager;
+
+    // To overcome circular dependency issue
+    @Lazy @Autowired @Setter
+    private SecurityUtils securityUtils;
+
+    public void logOverrideAudit(HttpServletRequest request) {
+        String roleAssignmentUrl = request.getHeader(roleAssignmentUrlManager.getUrlHeaderName());
+        String dataStoreUrl = request.getHeader(dataStoreUrlManager.getUrlHeaderName());
+
+        if (!isLogRequired(roleAssignmentUrl, dataStoreUrl)) {
+            return;
+        }
+
+        ObjectMapper mapper = new ObjectMapper();
+        ObjectNode root = mapper.createObjectNode();
+
+        String path = request.getRequestURI();
+        root.put("path", path);
+
+        if (roleAssignmentUrl != null && !roleAssignmentUrlManager.getHost().equals(roleAssignmentUrl)) {
+            root.put("role-assignment-url", roleAssignmentUrl);
+        }
+
+        if (dataStoreUrl != null && !dataStoreUrlManager.getHost().equals(dataStoreUrl)) {
+            root.put("data-store-url", dataStoreUrl);
+        }
+
+        root.put("hmctsServiceName", getServiceName(request.getHeader(SERVICE_AUTHORIZATION)));
+
+        if (isRequestWithBody(request)) {
+            root.put("requestBody", getRequestBody(request));
+        }
+
+        root.put("requestTimestamp", LocalDateTime.now(ZoneId.of("UTC")).toString());
+        root.put("user-id", securityUtils.getUserId());
+
+        if (path.startsWith("/linkedHearingGroup")) {
+            String groupId = getAttributeId(request, "id");
+            saveLinkedHearingStatusAudit(groupId, root);
+        } else {
+            String hearingId = getAttributeId(request, "id");
+            saveHearingStatusAudit(hearingId, root);
+        }
+    }
+
+    private void saveLinkedHearingStatusAudit(String groupId, ObjectNode root) {
+        LinkedHearingStatusAuditEntity auditEntity = new LinkedHearingStatusAuditEntity();
+        auditEntity.setHmctsServiceId("n/a");
+        auditEntity.setLinkedGroupId(groupId);
+        auditEntity.setLinkedGroupVersion("n/a");
+        auditEntity.setLinkedHearingEvent("n/a");
```

Comment:
we would know the hearing event in question at this point ie create-linked_hearing-request, update-linked_hearing-request or delete-linked_hearing-request based on the endpoint called

Reply:
Changed to the same as in `HearingStatusAudit` - OVERRIDE_URL event.